src/httpmanager.h

```diff
@@ -62,6 +62,9 @@ class HTTPManager
 public:
 	void GET(const char* pszUrl, CompletedCallback callback, std::vector<HTTPHeader>* headers = nullptr);
 	void POST(const char* pszUrl, const char* pszText, CompletedCallback callback, std::vector<HTTPHeader>* headers = nullptr);
+	void PUT(const char* pszUrl, const char* pszText, CompletedCallback callback, std::vector<HTTPHeader>* headers = nullptr);
+	void PATCH(const char* pszUrl, const char* pszText, CompletedCallback callback, std::vector<HTTPHeader>* headers = nullptr);
+	void DELETE(const char* pszUrl, const char* pszText, CompletedCallback callback, std::vector<HTTPHeader>* headers = nullptr);
```

Comment:
You're gonna need to pick a different name for this function. Apparently `DELETE` is already defined as a macro in Windows build environments, leading to Windows builds failing.

Reply:
If `DELETE` is renamed, the other should probably be renamed as well for consistency's sake? We could also just have a single method that takes the HTTP method as a parameter (either by creating our own enum or using valve's).

Reply:
Whatever works, `GET`/`POST` were only split up because of the extra param. Do be mindful of making compatibility breaks with the existing functions though, they are heavily used in server-specific forks.